assets/js/modules/analytics-4/components/dashboard/ActivationBanner/index.js

```diff
@@ -82,15 +157,47 @@ export default function ActivationBanner() {
 		}
 
 		return { dismissOnCTAClick: false };
-	}, [ returnToSetupStep, step ] );
+	}, [
+		step,
+		returnToSetupStep,
+		hasSetupBannerError,
+		setupBannerErrors,
+		setupBannerSelectors,
+		dispatch,
+	] );
 
 	if ( ! uaConnected || ga4Connected ) {
 		return null;
 	}
 
+	// Show unique errors.
+	const errorNotice =
+		hasSetupBannerError &&
+		Object.values( setupBannerErrors )
+			.reduce( ( acc, error ) => {
+				if (
+					! acc.some(
+						( err ) =>
+							err.code === error.code &&
+							err.message === error.message
+					)
+				) {
+					acc.push( error );
+				}
+
+				return acc;
```

Comment:
I would inverse the flow here and return the unmodified accumulator early if there's no changes to be made, and then return a modified one if there is a change to make, instead of mutating the accumulator and returning that…

```suggestion
				if (
					acc.some(
						( err ) =>
							err.code === error.code &&
							err.message === error.message
					)
				) {
					return acc;
				}

				return [ ...acc, error ];
```

Reply:
Very valid, addressed, thank you!